### ISSUE TITLE
Nvim doc text disappears

### ISSUE THREAD

**Issue by wwwsensor** [nvim-treesitter/nvim-treesitter#6989]
### Describe the bug

When Treesitter replaces Vim regex-based syntax highlighting, some text disappears.
Example:
`:h stl` => `*'statusline'* *'stl'* *E540* *E542*`

### To Reproduce

1. Deploy my [Neovim config](https://codeberg.org/profitz/nvim).
2. Run `:h stl`.

### Expected behavior

Text shouldn't disappear.

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v18.19.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `clang` executable found. Selected from { "clang" }
  Version: clang version 17.0.6
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "6.6.42_1",
  sysname = "Linux",
  version = "#1 SMP PREEMPT_DYNAMIC Fri Jul 26 12:44:34 UTC 2024"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - regex               ✓ . . . .
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~
```

### Output of `nvim --version`

```text
NVIM v0.10.1
Build type: None
LuaJIT 2.1.1720049189
Run "nvim -V1 -v" for more info
```

### Additional context

I use Void Linux glibc and I have tree-sitter CLI installed locally, even though Neovim doesn't detect it. Important to notice that I isolated individually one by one my plugins so I detect which one was causing this probem, it seems it was TS.

**Comment by clason** [comment]
You mean the asterisks? That's intended (help files use a lot of conceal).

**Comment by wwwsensor** [comment]
No, the whole text disappears. I'm obviously not talking only about the asteriks.

**Comment by clason** [comment]
I obviously have no idea if your report is so vague. 

1. Check if that happens with `nvim --clean` (the vimdoc parser is bundled and used by default in Neovim).
2. If it doesn't, make sure it's not your colorscheme ( `:color default` or `:color habamax` will work).

**Comment by xebcam** [comment]
I had the same issue. It's solved on the latests commit [d13f018](https://github.com/nvim-treesitter/nvim-treesitter/commit/d13f0183ba93f9b5aba7359146f294abfffff9c3)
